pkg/k8shandler/kibana/reconciler.go

```diff
@@ -0,0 +1,698 @@
+package kibana
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	kibana "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/pkg/constants"
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/util/retry"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	annotationOauthSecretUpdatedAt = "logging.openshift.io/oauthSecretUpdatedAt"
+	kibanaServiceAccountName       = "kibana"
+	kibanaOAuthRedirectReference   = "{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"kibana\"}}"
+)
+
+var (
+	kibanaServiceAccountAnnotations = map[string]string{
+		"serviceaccounts.openshift.io/oauth-redirectreference.first": kibanaOAuthRedirectReference,
+	}
+)
+
+func ReconcileKibana(requestCluster *kibana.Kibana, requestClient client.Client, proxyConfig *configv1.Proxy) error {
+	clusterKibanaRequest := ClusterKibanaRequest{
+		client:  requestClient,
+		cluster: requestCluster,
+	}
+
+	if clusterKibanaRequest.cluster == nil {
+		return nil
+	}
+
+	if err := clusterKibanaRequest.CreateOrUpdateServiceAccount(kibanaServiceAccountName, &kibanaServiceAccountAnnotations); err != nil {
+		return err
+	}
+
+	if err := clusterKibanaRequest.RemoveClusterRoleBinding("kibana-proxy-oauth-delegator"); err != nil {
```

Comment:
Remove?

Reply:
This is part of the `reconcileVisualization` logic from the CLO

Reply:
@jcantrill can you remember why we did this? was this because we changed how we did the  oauth stuff?

Reply:
Maybe as described https://github.com/openshift/elasticsearch-operator/pull/214/files/5687534d8037f668f05b61c063422d63383d2049#r370741776 because of https://github.com/openshift/cluster-logging-operator/pull/234